### PR TITLE
Curie update details

### PR DESCRIPTION
**Pull request by isabellewei**
- remove `--legacy` flag from Foundry commands
- update timeline and versions
- add audit link

### CODE REVIEW

File: src/content/docs/en/developers/guides/contract-deployment-tutorial.mdx [hunk 1]
```diff
@@ -43,7 +43,7 @@ The Scroll Sepolia Testnet allows anyone to deploy a smart contract on Scroll. I
 1. Clone the repo:
 
    ```shell
-   git clone https://github.com/scroll-tech/scroll-guides.git
```
Comment (dghelm) [inline src/content/docs/en/developers/guides/contract-deployment-tutorial.mdx]: Not sure why we're dropping the clone step here. (and there's a stray "c")

File: src/content/docs/en/developers/developer-quickstart.mdx [hunk 2]
```diff
@@ -113,7 +113,7 @@ const config: HardhatUserConfig = {
 To deploy using the Scroll Sepolia Testnet Public RPC, run:
 
 ```bash
-forge create ... --rpc-url=https://sepolia-rpc.scroll.io/ --legacy
```
Comment (dghelm) [inline src/content/docs/en/developers/developer-quickstart.mdx:116]: Do we need to wait until Alloy is updated? Can we test this and confirm?

**Review by dghelm — COMMENTED** [review]
Few notes. Also, it might be good to replace "PR ####" links in the Contract Changes section with labels that describe the PR (so readers don't have to open every one). Similar to Bridge Upgrade Code Changes section.

File: src/content/docs/en/developers/developer-quickstart.mdx (hunk 2 above)
Comment (isabellewei) [reply src/content/docs/en/developers/developer-quickstart.mdx:116]: I tested this on Scroll Sepolia and it did work already!

**Comment by isabellewei** [comment]
Added titles to PRs

**Review by dghelm — APPROVED** [review]

**Pull request merged**